### ISSUE TITLE
wash shows lots of unexpected errors when finding a file across all s3 buckets

### ISSUE THREAD

**Issue by timidri** [puppetlabs/wash#126]
**Describe the bug**
When issuing a `find` command across all s3 buckets, ~the command errors out and the wash server exits with a stack trace.~ the command produces lots of strange errors like
```
could not get children of ./int-resources/Developer/Applications/Dashcode.app/Contents/Frameworks/CYKit.framework/Versions/A/Resources/Parts/Text.part/es.lproj: puppetlabs.wash/entry-not-found: Could not find entry default/resources/s3/int-resources/Developer/Applications/Dashcode.app/Contents/Frameworks/CYKit.framework/Versions/A/Resources/Parts/Text.part/es.lproj: SerializationError: failed to decode REST XML response
	status code: 200, request id: 9EF65C82259987FA
caused by: read tcp 192.168.1.109:55137->52.218.233.201:443: read: connection reset by peer
```

**To Reproduce**
1. run `wash server ~/wash_mnt --loglevel trace` 
2. In a different terminal. run `find ~/wash_mnt/aws/default/resources/s3 -name index.yaml -ls`
3. After a couple of minutes, the command errors out like so:
```
find: /Users/dimitri.tischenko/wash_mnt/aws/default/resources/s3/2015.puppetconf.com-logs/logs: Device not configured
find: fts_read: Device not configured
```

**Expected behavior**
I expect the find command to print all the files called `index.yaml`.

**Logs**
The wash server log can be found here: https://gist.github.com/timidri/fce2dd0fcddecf8ea3257c61a2dbd642

**Environment (please complete the following information):**
 - OS: [macOS 10.14.3]
 - Version/Build number [153df44f49bb197a173b5fe80285ab43cc630528] (aws-round2)
 - Cloud target [AWS]

**Comment by MikaelSmith** [comment]
107k objects is a lot. I'm not exactly sure how we want to handle that. At the least we shouldn't request information about each item; I'm not even sure we should attempt to represent all of them in the filesystem.

**Comment by ekinanp** [comment]
This probably means that we'll need to implement `wash find`

**Comment by timidri** [comment]
This begs the question whether the strategic way forward is to work "inside wash's world" and reimplement all the needed commands there instead of mounting the filesystem and hitting FUSE's limits?

**Comment by MikaelSmith** [comment]
Mind trying this again? I think it will work better now.

**Comment by ekinanp** [comment]
One caveat is that `wash find` (currently) doesn't support any actions, so you'll have to pipe the outputted objects into `xargs` to get the desired effect.

**Comment by MikaelSmith** [comment]
I believe this is fixed by recent improvements, going to close it for lack of activity.

**Comment by timidri** [comment]
@MikaelSmith I've finally gotten around to test this again.

After running the command:

```
~/Library/Caches/wash/mnt943068102/aws/default/resources/s3  wash find . -name index.yaml
```
I am getting a different error now, see gist: https://gist.github.com/timidri/e9b7b318f7d238259d9dfe0f461ee09c

The command took 158:25 to run.

**Comment by MikaelSmith** [comment]
Hmm. So finding a file by name in S3 seems like it would be useful. But querying everything in S3 is going to be expensive (both in time and possibly dollars). So unless there's an API that we could leverage, I don't think this is where I want to spend effort at the moment.

This may be improved if we move some of the find logic into the wash daemon, which we've discussed. I suspect in this instance caching is expiring and making it take longer than it would otherwise.

**Comment by timidri** [comment]
@MikaelSmith I've done a brief search but I don't think there is an API which allows a more efficient search for object keys across multiple buckets. Totally agree enabling this is probably not the best use of resources at the moment.

**Comment by MikaelSmith** [comment]
We should understand whatever's causing
```
could not get children of ./int-resources/Developer/Applications/Dashcode.app/Contents/Frameworks/CYKit.framework/Versions/A/Resources/Parts/Text.part/es.lproj: puppetlabs.wash/entry-not-found: Could not find entry default/resources/s3/int-resources/Developer/Applications/Dashcode.app/Contents/Frameworks/CYKit.framework/Versions/A/Resources/Parts/Text.part/es.lproj: SerializationError: failed to decode REST XML response
	status code: 200, request id: 9EF65C82259987FA
caused by: read tcp 192.168.1.109:55137->52.218.233.201:443: read: connection reset by peer
```
and
```
could not get children of ./int-resources/Network: puppetlabs.wash/entry-not-found: Could not find entry default/resources/s3/int-resources/Network: RequestTimeTooSkewed: The difference between the request time and the current time is too large.
	status code: 403, request id: C4D073ABB92AF462, host id: jsLNEnFm2KbJenyGmDFu5ALISqyubGyRKj0J72/j5sZ+NatzUzuee+TqxA/QwPgSRfh1RIyT+5I=
```

Failure to decode REST XML response in particular seems like a problem.

Also, the error printed should contain as much information as we have to debug what's failing. If there's more available, we should expose it.

**Comment by ekinanp** [comment]
I haven't been able to replicate this. Turns out I have access to the `int-resources` bucket (and probably all the buckets that the invocation here was searching through).

```
wash aws/demo_two/resources/s3 ❯ find int-resources/ -name index.yaml
```

That worked fine (although it took a while). I've currently been running
```
wash aws/demo_two/resources/s3 ❯ find . -name index.yaml
./dimitri-chart-repo/index.yaml
...
```

and after almost 5 hours, I've yet to see the error messages. I've cancelled the invocation to avoid making any more API requests. Note that I also checked the invocation's activity log and everything looked fine.

RE: The errors, I think the XML one is generated by https://github.com/aws/aws-sdk-go/blob/master/private/protocol/restxml/restxml.go#L52-L69 while https://forums.aws.amazon.com/thread.jspa?threadID=12628 might give more insight into the RequestTimeTooSkewed error.
